File: app/src/main/java/com/college/app/adapter/EventsAdapter.java
package com.college.app.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.college.app.databinding.EventItemBinding;
import com.college.app.utils.Event;

import java.util.List;

public class EventsAdapter extends RecyclerView.Adapter<EventsAdapter.Item> {

    List<Event> eventList;

    public EventsAdapter(List<Event> eventList) {
        this.eventList = eventList;
    }

    @NonNull
    @Override
    public Item onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        EventItemBinding eventItemBinding = EventItemBinding.inflate(layoutInflater, parent, false);
        return new Item(eventItemBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull Item holder, int position) {
        holder.eventItemBinding.eventName.setText(eventList.get(position).getEventName());
        holder.eventItemBinding.eventDesc.setText(eventList.get(position).getEventDesc());
        holder.eventItemBinding.eventDate.setText(eventList.get(position).getEventDate());
    }

    @Override
    public int getItemCount() {
        return eventList.size();
    }

    public static class Item extends RecyclerView.ViewHolder {
        EventItemBinding eventItemBinding;

        public Item(EventItemBinding eventItemBinding) {
            super(eventItemBinding.getRoot());
            this.eventItemBinding = eventItemBinding;
        }
    }
}

File: app/src/main/java/com/college/app/adapter/EventsAdapter.kt
package com.college.app.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.college.app.databinding.EventItemBinding
import com.college.app.utils.Event

class EventsAdapter(var eventList: List<Event>?) : RecyclerView.Adapter<EventsAdapter.Item>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Item {
        val layoutInflater = LayoutInflater.from(parent.context)
        val eventItemBinding = EventItemBinding.inflate(layoutInflater, parent, false)
        return Item(eventItemBinding)
    }

    override fun onBindViewHolder(holder: Item, position: Int) {
        holder.eventItemBinding.eventName.text = eventList!![position].eventName
        holder.eventItemBinding.eventDesc.text = eventList!![position].eventDesc
        holder.eventItemBinding.eventDate.text = eventList!![position].eventDate
    }

    override fun getItemCount(): Int {
        return eventList!!.size
    }

    class Item(var eventItemBinding: EventItemBinding) : RecyclerView.ViewHolder(
        eventItemBinding.root
    )
}